src/static_canvas.class.js

```diff
@@ -569,6 +569,13 @@
      */
     _onObjectAdded: function(obj) {
       this.stateful && obj.setupState();
+      if (obj.canvas && obj.canvas !== this) {
+        /* _DEV_MODE_START_ */
+        console.warn('fabric.Canvas: trying to add an object that belongs to a different canvas.\n' +
+          'Resulting to default behavior: removing object from previous canvas and adding to new canvas');
+        /* _DEV_MODE_END_ */
+        obj.canvas.remove(obj);
+      }
```

Comment:
i think this is totally fair, because the old reference in the canas would be just a broken reference.

i used this a couple of times to render objects in different canvases for copy purposes, the thing i was doing was to swap the canvas reference manually, while now i should go through the add/remove process and avoid extra renders.

Reply:
Do we want to check for duplicates as well? I think it's damaging for perf and I don't see too much advantage for it

Reply:
if we do we can do in the dev mode braket and remove it in prod build.